src/main/java/killjoy/task/Task.java

```diff
@@ -0,0 +1,63 @@
+package killjoy.task;
+
+/**
+ * Represents the Task class of the KillJoy application.
+ * Contains methods to create and manage tasks.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType = null;
+
+    /**
+     * Represents the type of task.
+     */
+    public enum TaskType {
+        TODO,
+        DEADLINE,
+        EVENT
+    }
+
+    public Task(String description, TaskType taskType) {
+        this.description = description;
+        this.isDone = false;
+        this.taskType = taskType;
+    }
+
+    /**
+     * Changes the status of the task.
+     */
+    public void changeStatus() {
+        if (this.isDone) {
+            this.isDone = false;
+        } else {
+            this.isDone = true;
+        }
+    }
```

Comment:
i like the naming but perhaps could give greater details by naming it as changeTaskStatus instead?